### PR TITLE
fix(shadowtable): bad sql grammer when table column's default value is a string type in MySQL

### LINKED ISSUE #2379: [Bug]: The sql statement generated by the shadow table task does not put quotes on the default field
### ODC version

ODC424bp1

### OB version

~

### What happened?

```
create table test_char(`col12` char(10) NOT NULL DEFAULT '中文字段' COMMENT '第十二列');
```
sql statements generated by the shadow table task:
```
create table test_char(`col12` char(10) NOT NULL DEFAULT  中文字段  COMMENT '第十二列');
```

### What did you expect to happen?

Execution report syntax error.

### How can we reproduce it (as minimally and precisely as possible)?

Follow what happened.

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by MarkPotato777**
#### What type of PR is this?

type-bug

#### What this PR does / why we need it:
When the original table column's default is a string type, the generated ddl does not quote the default value which causes to sql execution failed.

#### Which issue(s) this PR fixes:

Fixes #2379 

#### Special notes for your reviewer:

This issue was introduced by https://github.com/oceanbase/odc/pull/1401

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by yhilmare — DISMISSED** [review]
lgtm

**Review by PeachThinking — DISMISSED** [review]
LGTM

File: server/odc-common/src/main/java/com/oceanbase/odc/common/util/StringUtils.java [hunk 1]
```diff
@@ -163,6 +168,31 @@ public static String unquoteSqlValue(final String str, final char wrapChar, fina
         return unescapeUseDouble(unwrap, escapeChars);
     }
 
+    public static void quoteColumnDefaultValues(DBTable table) {
```
Comment (yhilmare) [inline server/odc-common/src/main/java/com/oceanbase/odc/common/util/StringUtils.java:171]: we may rename it: quoteColumnDefaultValuesForMysql

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by PeachThinking — APPROVED** [review]
LGTM

**Pull request merged**